src/as2fm/scxml_converter/scxml_entries/xml_utils.py

```diff
@@ -93,30 +97,45 @@ def read_value_from_xml_child(
     xml_child = xml_tree.findall(child_tag)
     if xml_child is None or len(xml_child) == 0:
         if not none_allowed:
-            print(f"Error: reading from {xml_tree.tag}: Cannot find child '{child_tag}'.")
+            log_error(
+                xml_tree,
+                f"Error: reading from {xml_tree.tag}: Cannot find child '{child_tag}'.",
+            )
         return None
     if len(xml_child) > 1:
-        print(f"Error: reading from {xml_tree.tag}: multiple children '{child_tag}', expected one.")
+        log_error(
+            xml_tree,
+            f"Error: reading from {xml_tree.tag}: multiple children '{child_tag}', expected one.",
+        )
         return None
     tag_children = [child for child in xml_child[0] if not is_comment(child)]
     n_tag_children = len(tag_children)
     if n_tag_children == 0 and str in valid_types:
         # Try to read the text value
         text_value = xml_child[0].text
         if text_value is None or len(text_value) == 0:
-            print(f"Error: reading from {xml_tree.tag}: Child '{child_tag}' has no text value.")
+            log_error(
+                xml_tree,
+                f"Error: reading from {xml_tree.tag}: Child '{child_tag}' has no text value.",
+            )
             return None
         return text_value
     if n_tag_children > 1:
-        print(f"Error: reading from {xml_tree.tag}: Child '{child_tag}' has multiple children:")
+        log_error(
+            xml_tree,
+            f"Error: reading from {xml_tree.tag}: Child '{child_tag}' has multiple children:",
+        )
         for child in tag_children:
-            print(f"\t- {child.tag}")
+            log_error(xml_tree, f"\t- {child.tag}")
```

Comment:
This was supposed to be a longer error message, containing also the list of children... Maybe we can simply remove the for loop and add the n. of children in the error message